src/cascadia/TerminalCore/TerminalDispatch.cpp

```diff
@@ -474,7 +474,19 @@ bool TerminalDispatch::DoConEmuAction(const std::wstring_view string) noexcept
     {
         if (parts.size() >= 2)
         {
-            return _terminalApi.SetWorkingDirectory(til::at(parts, 1));
+            const auto path = til::at(parts, 1);
+            // The path should be surrounded with '"' according to the documentation of ConEmu.
+            // An example: 9;"D:/"
+            if (path.at(0) == L'"' && path.at(path.size() - 1) == L'"' && path.size() >= 3)

```

Comment:
Note that this is actually making OSC 9;9 even more Win32-specific. Since `"` is an invalid dir name on WIndows, but is a valid dir name on Linux.

Reply:
To be specific, when we see a path provided by OSC 9;9 , for example, `"D:"` on Windows, the actual path is guaranteed to be `D:` because neither `"` nor `:` is valid character in dir names on Windows. But on Linux, you can create a dir that's literally named `"D:"` by `mkdir "\"D:\""`.

I can't blame ConEmu for designing the sequence this way since it's a Win32 native application at the very beginning. Just to point it out.

Also CC @Maximus5 for awareness.